wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/service/TextDocumentService.java

```diff
@@ -306,6 +310,21 @@ public DocumentHighlight documentHighlight(TextDocumentPositionParamsDTO positio
     	}
     	return null;
     }
+    
+    @POST
+    @Path("codeAction")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+	public List<? extends Command> codeAction(CodeActionParamsDTO params)
+					throws LanguageServerException, InterruptedException, ExecutionException {
+    	params.getTextDocument().setUri(prefixURI(params.getTextDocument().getUri()));
```

Comment:
Pretty much every other method in this class does the exact same thing. I did not want to introduce a new pattern here. If that pattern is not desired, we should fix it, but that's outside of the scope of this PR.